### PR TITLE
Core/GameObject: Implement DespawnAtAction

### PR DESCRIPTION
**Pull request by Killyana**
**Changes proposed:**

-    Fix the logic used by IsDespawnAtAction for chest and goobers
-    Keep Goobers and chests that are summoned, spawned until the timer expire, like Romantic Basket case

**Target branch(es):** 3.3.5

- [x] 3.3.5
- [ ] master

**Issues addressed:** #15730

**Tests performed:** Yes

**Known issues and TODO list:** (add/remove lines as needed)

- [ ]  This works only If I add: m_respawnTime > 0 :
``` cpp
 if (m_respawnTime > 0 &&  !GetGOInfo()->IsDespawnAtAction())
```
But this will exclude DB spawns, I can't figure out why.

### CODE REVIEW

**Comment by Kittnz** [comment]
Might want to add a random or between option

File: src/server/game/Entities/GameObject/GameObjectData.h [hunk 1]
```diff
@@ -477,7 +477,7 @@ struct GameObjectTemplate
         }
     }
 
-    bool GetDespawnPossibility() const                      // despawn at targeting of cast?
+    bool CannotBeUsedUnderImmunity() const                      // The object cannot be use under Immunity like divine shield -- to be implemented
```
Comment (illfated) [inline src/server/game/Entities/GameObject/GameObjectData.h]:
```suggestion
    bool CannotBeUsedUnderImmunity() const                  // The object cannot be used under Immunity like divine shield -- to be implemented
```

**Comment by jackpoz** [comment]
Need help rebasing ?

**Comment by illfated** [comment]
Looks like it. Second time in a short row.

**Comment by Killyana** [comment]
I don't know how to fix the remaining issue, no need to have the PR open for years.

**Comment by jackpoz** [comment]
I can debug it when I have time if you need, just let me know

**Comment by Killyana** [comment]
It will be nice.

**Pull request closed without merging**